src/coreclr/jit/emitarm64.cpp

```diff
@@ -6244,14 +6353,21 @@ void emitter::emitIns_R_I(instruction ins,
     assert(canEncode);
     assert(fmt != IF_NONE);
 
-    instrDesc* id = emitNewInstrSC(attr, imm);
+    // Instructions with optional shifts need larger instrDesc to store state
+    instrDesc* id = optionalShift ? emitNewInstrCns(attr, imm) : emitNewInstrSC(attr, imm);
```

Comment:
please double check the TP cost for this.

Reply:
Sure thing. [This](https://dev.azure.com/dnceng-public/public/_build/results?buildId=534113&view=ms.vss-build-web.run-extensions-tab) is from an older run, but TP shouldn't have changed since my last push. There is a TP impact, but it's quite small, and I think it only comes from the additional branch from checking `optionalShift`. The larger `instrDesc` should only affect SVE instructions as of writing, as `optionalShift` is only true for the new encodings added in this PR.

Reply:
There are 2 compares (probably c++ eliminates redundant comparison) happening for non-sve and that is increasing the `tp`. May be in future, we should have `emitIns_SVE_R_I()` .

Can you change this to so the frequent branch is always true.

```c++
instrDesc* id = nullptr;

if (!optionalShift)
{
 id = ...
 ...
}
else
{
 ...
  id = emitNewInstrCns(attr, imm);
  id->idOptionalShift(hasShift);
}
```

Reply:
The [updated diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=535995&view=ms.vss-build-web.run-extensions-tab) look the same.

Reply:
yeah, we might eventually have to move to `emitIns_sve_`, but lets hold off on that for now.